Dd/#3/in memory crud expense

daaariaa opened the pull request:
CRUD operations for managing expenses

daaariaa committed:
first-commit

daaariaa committed:
second-commit

daaariaa committed:
third-commit

daaariaa committed:
fourth-commit

Nikolas5516 commented on backend/src/main/java/cloudflight/integra/backend/entity/validation/ExpenseValidation.java:
```diff
@@ -0,0 +1,31 @@
+public class ExpenseValidation implements Validator<ExpenseDto> {
+    @Override
+    public void validateException(ExpenseDto expenseDto) throws ValidationExceptionExpense {
+        String errors = "";
```
Try to add the errors in a list

daaariaa committed:
fifth-commit

the-coding-cloud commented on backend/src/main/java/cloudflight/integra/backend/controller/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,87 @@
+    public ResponseEntity<Map<String, Object>> handleGeneral(Exception ex) {
+        log.error("Unexpected error: ", ex);
+        Map<String, Object> body = new HashMap<>();
+        body.put("timestamp", LocalDateTime.now());
```
small improvement - extract the response body setting in another method

the-coding-cloud reviewed (commented):
great job! make sure to document the quick test

daaariaa committed:
Refactor: extract error response body construction in exception handler

- Moved response body from general exception handler to a separate method
- Improves code readability and allows reuse for other exception handlers

daaariaa committed:
Refactor: extract error response body construction in exception handler

- Moved response body from general exception handler to a separate method
- Improves code readability and allows reuse for other exception handlers

daaariaa committed:
Refactor: extract error response body construction in exception handler

- Moved response body from general exception handler to a separate method
- Improves code readability and allows reuse for other exception handlers

daaariaa committed:
#3 Refactor global exception handling
